### PR TITLE
txnbuild: full end-to-end SDK demo

### LINKED ISSUE #1070: txnbuild: Move examples to use new exp/horizonclient
Examples currently use the old client. Move over to the new API in preparation for release.

### PR DESCRIPTION
**Pull request by ire-and-curses**
This PR adds a command line demo tool that shows some real-world usage of the new Go SDK (`horizonclient` + `txnbuild`). This code is intended to be a teaching tool for new users. This code will also be used as a basis for worked examples on the Stellar developers site.

Due to lack of time, not all operations are covered, and it is definitely a little rough around the edges. Nevertheless this is a fairly simple set of working examples that should help a new SDK developer find their way around the basics of the API.

The code demonstrates correct usage of friendbot, using an initial account to fund others, and merging accounts back to friendbot when done.

To try the demo out, checkout the branch and run `go build` from the `exp/txnbuild/cmd/demo` directory. Then run the binary: `./demo`.

Please ignore the file `txnbuild_examples`. I plan to remove this in a separate PR when I add examples to the GoDoc.

Closes #1070.

### CODE REVIEW

**Comment by poliha** [comment]
Hitting a panic 
```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/stellar/go/exp/txnbuild/cmd/demo/operations.InitKeys(0x4, 0x0, 0x0, 0x0)
	... stellar/go/exp/txnbuild/cmd/demo/operations/demo.go:49 +0xa7a
```

File: exp/txnbuild/cmd/demo/operations/demo.go [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+	log.Info("Address:", pair.Address())
+
+	return Account{
+		Seed:    pair.Seed(),
+		Address: pair.Address(),
+		Keypair: pair,
+	}
+}
+
+func loadAccounts(client *horizonclient.Client, accounts []Account) []Account {
```
Comment (poliha) [inline exp/txnbuild/cmd/demo/operations/demo.go]: docstring

File: exp/txnbuild/cmd/demo/operations/demo.go [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+	}
+
+	tx := txnbuild.Transaction{
+		SourceAccount: source,
+		Operations:    []txnbuild.Operation{&accountMerge},
+		Timebounds:    txnbuild.NewTimeout(300),
+		Network:       network.TestNetworkPassphrase,
+	}
+
+	txeBase64, err := tx.BuildSignEncode(signer.Keypair)
```
Comment (poliha) [inline exp/txnbuild/cmd/demo/operations/demo.go]: this is repeated a lot, maybe move it to function?
``` 	
txeBase64, err := tx.BuildSignEncode(signer.Keypair)
	if err != nil {
		return "", errors.Wrap(err, "couldn't serialise transaction")
	}

```

Comment (ire-and-curses) [reply exp/txnbuild/cmd/demo/operations/demo.go]: Thanks - I realised I could simplify this

**Pull request merged**